### PR TITLE
fix: leverage memo to avoid re-renders

### PR DESCRIPTION
**Pull request by rebelchris**
### Describe what this PR does
- Remove effect/state in favor of memo.

## Events

Did you introduce any new tracking events?
Don't forget to update the [Analytics Taxonomy sheet](https://docs.google.com/spreadsheets/d/18Lv7zXges9QfVX5VYL1a-Hyl0e1sQ3sLr0OK8YZWKXI/edit#gid=0)

Log the new/changed events below:

| Type   | event_name  | value |
|--------|-------------|-------|
| Change/New | event name  | extra: { ... } |

### On those affected packages:
- [ ] Have you done sanity checks in the webapp?
- [ ] Have you done sanity checks in the extension?
- [ ] Does this not break anything in companion?

### Did you test the modified components media queries?
- [ ] MobileL (420px)
- [ ] Tablet (656px)
- [ ] Laptop (1020px)

#### Did you test on actual mobile devices?
- [ ] iOS (Chrome and Safari)
- [ ] Android

WT-2056 #done

### CODE REVIEW

**Review by sshanzel — APPROVED** [review]
From 4 hooks to one 🥳

File: packages/webapp/components/layouts/ProfileLayout/index.tsx [hunk 1]
```diff
@@ -93,15 +84,19 @@ export default function ProfileLayout({
...
-      setGithubHandle(sanitizeOrNull(purify, profile.github));
-      setHashnodeHandle(sanitizeOrNull(purify, profile.hashnode));
-      setPortfolioLink(sanitizeOrNull(purify, profile.portfolio));
-    }
-  }, [profile]);
+  const { twitterHandle, githubHandle, hashnodeHandle, portfolioLink } =
+    useMemo(() => {
+      if (typeof window === 'undefined') {
+        return {};
+      }
```
Comment (capJavert) [inline packages/webapp/components/layouts/ProfileLayout/index.tsx:91]: Maybe also just return if `profile` is undefined, no need to call `sanitizeOrNull` when stuff is undefined?

**Review by capJavert — APPROVED** [review]
Nice!

File: packages/webapp/components/layouts/ProfileLayout/index.tsx (hunk 1 above)
Comment (rebelchris) [reply packages/webapp/components/layouts/ProfileLayout/index.tsx]: Good point, adjusted now.

**Review by capJavert — APPROVED** [review]

**Pull request merged**